src/plugins/media/media.js

```diff
@@ -99,70 +169,74 @@
     };
 
     enhanceMedia = function () {
-        var steps, i;
+        var steps, stepElement, i;
         enhanceMediaNodes();
-        steps = document.getElementsByClassName('step');
-        for (i = 0; i < steps.length; i++) {
-            gc.addEventListener(steps[i], 'impress:stepenter', onStepenter);
-            gc.addEventListener(steps[i], 'impress:stepleave', onStepleave);
+        steps = document.getElementsByClassName("step");
+        for (i = 0; i < steps.length; i += 1) {
+            stepElement = steps[i];
+            
+            gc.addEventListener(stepElement, "impress:stepenter", onStepenter);
+            gc.addEventListener(stepElement, "impress:stepleave", onStepleave);
         }
     };
 
     onStepenter = function (event) {
-        var media, play, i;
+        var stepElement, media, mediaElement, i, onConsolePreview, onConsoleSlideView;
         if ((!event) || (!event.target)) {
             return;
         }
+        stepElement = event.target;
+        
         removeMediaClasses();
-        media = event.target.querySelectorAll('audio, video');
-        for (i = 0; i < media.length; i++) {
-            play = media[i].dataset.mediaAutoplay;
-            if (play !== undefined && play !== "false") {
-                media[i].play();
+        
+        media = stepElement.querySelectorAll("audio, video");
+        for (i = 0; i < media.length; i += 1) {
+            mediaElement = media[i];
+            
+            // Autoplay when (maybe inherited) autoplay setting is true, 
```

Comment:
Could you try to move all impressConsole related stuff to a separate helper function?

* The onConsoleSlideView part you should be able to handle completely inside that function
* The function could return true if we are in fact in any window that is part of impressConsole

This would simplify the code that remains here to something like:

    mediaElement = media[i]
    isConsole = onStepenterImpressConsole(mediaElement);
    if (getMediaAttribute(...) && !isConsole) {
        mediaElement.play()
    }